### PR TITLE
FIX APIv2 : Add exception when parser fail to avoid error 500 without error info

### PR DESCRIPTION
**Pull request by RomainJufer**
The method `save` in the serializers for import scan and re-import scan can fail if the file parsed is not of the type given by scan type. As a result, the client gets an error 500 without any error message. Therefore, this PR aim at solving that issue by raising an exception which is catched in the viewset and transformed into a ParseError that yields an HTTP error 400 with an error message.

Currently, I wrote a generic message since there is not error handling in the `import_parser_factory` that would yield a more specific error message. Let me know if you think that something else would be more appropriate.

### CODE REVIEW

**Review by madchap — APPROVED** [review]

**Comment by madchap** [comment]
@RomainJufer we're planning to release 1.10.1 soonish, would you mind rebasing off of `release/1.10.1` branch please? That way, we can include the fix there too. Thanks.

**Review by Maffooch — APPROVED** [review]
It's a little larger of a rebase than I was hoping. Will merge after rebase is complete.

File: dojo/api_v2/serializers.py [hunk 1]
```diff
@@ -1002,6 +1002,8 @@ def save(self, push_to_jira=False):
                                            data['scan_type'],)
         except ValueError:
             raise Exception('FileParser ValueError')
+        except:
+            raise Exception('Error while parsing the report, did you specify the correct scan type ?')
```
Comment (valentijnscholten) [inline dojo/api_v2/serializers.py:1006]: I think this hides the original exception? should it be `raise Exception('blabla', e)` or something?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

**Comment by madchap** [comment]
> It's a little larger of a rebase than I was hoping. Will merge after rebase is complete.

I am sure @RomainJufer will come up with some `git rebase --onto` magic and get this by like a breeze ;-)

File: dojo/api_v2/serializers.py (hunk 1 above)
Comment (valentijnscholten) [reply dojo/api_v2/serializers.py:1006]: Hmm just read that it was on purpose to catch it in the viewset. This was on our list for a long time to make error handling better, so thanks for doing this.

**Review by valentijnscholten — APPROVED** [review]

**Comment by valentijnscholten** [comment]
Or just leave it at dev. The hotfix shouldn't be too large and yours6 or more of an enhancement.

**Pull request merged**